app.go

```diff
@@ -461,6 +467,25 @@ func (app *App) StopTimeout() time.Duration {
 	return app.stopTimeout
 }
 
+func (app *App) visualizeDOT() DotGraph {
+	// Memoize the graph
+	if app.dotgraph != nil {
+		return *app.dotgraph
+	}
+	b := &bytes.Buffer{}
+	if err := dig.Visualize(app.container, b); err != nil {
+		b.WriteString(`digraph {
+  graph [];
+  
+  error  [shape=plaintext label="Error graphing container"];
```

Comment:
Let's include the error message contents.

Reply:
Also, can we the generation of the error-case graph into a function rather than inlining here.